modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/JavaDeclarationDelegateTestCase.java

```diff
@@ -871,7 +872,15 @@ private void assertTransactional(ExecutableComponentDeclaration declaration, boo
   }
 
   private void assertOutputType(OutputDeclaration output, MetadataType type, boolean isDynamic) {
-    assertThat(output.getType(), equalTo(type));
+    MetadataType outputType = output.getType();
+    String outputTypeId = ExtensionMetadataTypeUtils.getId(outputType).orElse(null);
+    String actualTypeId = ExtensionMetadataTypeUtils.getId(type).orElse(null);
+
+    if (outputTypeId != null && type instanceof ObjectType) {
+      assertThat(actualTypeId, equalTo(actualTypeId));
```

Comment:
why is this needed? How was output types changed and how does it relate to stereotypes?

Reply:
This is because the StereotypeTypeAnnotations are stateful and their state was being altered at an ExtensionFactory level enricher. This was incorrect and I fixed it. As a result, the types are not quite that equal because of that stateful type annotation. Because what this test is verifying is the declarer and not the extension itself (which is handcrafted and filled with inaccuracies) I considered this change to be benign.